pkg/daemon/ceph/client/osd.go

```diff
@@ -355,3 +355,15 @@ func OSDOkToStop(context *clusterd.Context, clusterInfo *ClusterInfo, osdID, max
 
 	return stats.OSDs, nil
 }
+
+// OsdPrimaryAffinity assigns primary-affinity (within range [0.0, 1.0]) to a specific OSD.
+func OsdPrimaryAffinity(context *clusterd.Context, clusterInfo *ClusterInfo, osdID int, affinity string) error {
+	logger.Infof("setting osd.%d with primary-affinity %q", osdID, affinity)
+	args := []string{"osd", "primary-affinity", fmt.Sprintf("osd.%d", osdID), affinity}
```

Comment:
Since what version of Ceph is this command available? We still support Nautilus/Octopus upstream

Reply:
Support for primary-affinity dates back to 2014